### PR TITLE
Remove unused STRING_SET Entity.

### PR DESCRIPTION
**Pull request by kylep-dremio**
STRING_SET and LIST are essentially similar, so remove STRING_SET in favour of the generic list.

---
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/506)

### CODE REVIEW

File: versioned/dynamodb/src/test/java/com/dremio/nessie/versioned/impl/condition/TestExpressions.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
   private final Entity av0 = Entity.ofBoolean(true);
   private final Entity av1 = Entity.ofBoolean(false);
   private final Entity av2 = Entity.ofString("mystr");
-  private final Entity set1 = Entity.ofStringSet("foo", "bar");
+  private final Entity set1 = Entity.ofList(Entity.ofString("foo"), Entity.ofString("bar"));
```
Comment (jacques-n) [inline versioned/dynamodb/src/test/java/com/dremio/nessie/versioned/impl/condition/TestExpressions.java:32]: I'm pretty sure this variable is used in various parts of the tests...

Comment (kylep-dremio) [reply versioned/dynamodb/src/test/java/com/dremio/nessie/versioned/impl/condition/TestExpressions.java:32]: It is - I altered it to be a list to ensure the tests would pass. There are 3 affected tests.

Comment (jacques-n) [reply versioned/dynamodb/src/test/java/com/dremio/nessie/versioned/impl/condition/TestExpressions.java:32]: The use of set was used in places that require a set in dynamo. You need to remove the use entirely.

**Review by kylep-dremio — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/projectnessie/nessie/506)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @jacques-n)

___
*[versioned/dynamodb/src/test/java/com/dremio/nessie/versioned/impl/condition/TestExpressions.java, line 32 at r1](https://reviewable.io/reviews/projectnessie/nessie/506#-MNB0Tv4CAEiAfp8pfwI-r1-32:-MNB_z8d0QHZXNT5ttCC:b-896fix) ([raw file](https://github.com/projectnessie/nessie/blob/15ee28efab68303a6a40be200e5ec2d61a1f0a3d/versioned/dynamodb/src/test/java/com/dremio/nessie/versioned/impl/condition/TestExpressions.java#L32)):*
<details><summary><i>Previously, jacques-n (Jacques Nadeau) wrote…</i></summary><blockquote>

The use of set was used in places that require a set in dynamo. You need to remove the use entirely.
</blockquote></details>

Done.

**Review by jacques-n — APPROVED** [review]

**Pull request merged**